clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -168,190 +175,194 @@ public Map<Errors, Integer> errorCounts(Throwable e) {
      * Factory method for getting a request object based on ApiKey ID and a version
      */
     public static RequestAndSize parseRequest(ApiKeys apiKey, short apiVersion, ByteBuffer buffer) {
-        int bufferSize = buffer.remaining();
-        return new RequestAndSize(doParseRequest(apiKey, apiVersion, buffer), bufferSize);
+        return parseRequest(apiKey, apiVersion, new ByteBufferAccessor(buffer));
```

Comment:
Do we need an overload that takes a ByteBuffer?  I found one case (see my other comment), but it could be easily fixed if `serialize()` returned `ByteBufferAccessor`.

Reply:
Done.